src/MonoTorrent/MonoTorrent.Client/ClientEngine.cs

```diff
@@ -468,7 +475,9 @@ private void LogicTick ()
                 uploadLimiter.UpdateChunks (Settings.MaximumUploadSpeed, TotalUploadSpeed);
             }
 
-            ConnectionManager.CancelPendingConnects ();
+            for (int i = 0; i < this.torrents.Count; i++)
+                ConnectionManager.CancelPendingConnects(this.torrents[i]);
```

Comment:
This change looks like it's broken. It should cause pending connects to be closed every time LogicTick is invoked rather than only when the timeout has been reached. This means pending connections will be closed between 0 and 500ms after the connection attempt is initiated. Do you recall what issue you were trying to solve with this change?